### PR TITLE
add exec/target constraints parameters to python toolchain macro

### PR DESCRIPTION
**Pull request by tek**
just copied the code from the CC toolchain, works fine

### CODE REVIEW

File: nixpkgs/nixpkgs.bzl [hunk 1]
```diff
@@ -890,6 +890,18 @@ def nixpkgs_cc_configure_deprecated(
...
+    if repository_ctx.attr.target_constraints == None and repository_ctx.attr.exec_constraints == None:
+        target_constraints = ["@platforms//cpu:x86_64"]
+        target_constraints.append("@platforms//os:{}".format(os))
+        exec_constraints = target_constraints
+    else:
+        target_constraints = list(repository_ctx.attr.target_constraints)
+        exec_constraints = list(repository_ctx.attr.exec_constraints)
+
+    exec_constraints.append("@io_tweag_rules_nixpkgs//nixpkgs/constraints:support_nix")
+
```
Comment (aherrmann) [inline nixpkgs/nixpkgs.bzl:904]: This is the same as in `nixpkgs_cc_configure`, correct? Could you factor this out into a function?

**Review by aherrmann — APPROVED** [review]
Thank you! Small comment.

CI is failing because the README me is outdated. The test failure should state how to fix it.

File: nixpkgs/nixpkgs.bzl (hunk 1 above)
Comment (tek) [reply nixpkgs/nixpkgs.bzl:904]: with pleasure!

File: nixpkgs/nixpkgs.bzl [hunk 2]
```diff
@@ -890,6 +890,18 @@ def nixpkgs_cc_configure_deprecated(
 
 def _nixpkgs_python_toolchain_impl(repository_ctx):
     cpu = get_cpu_value(repository_ctx)
+    os = {"darwin": "osx"}.get(cpu, "linux")
+
+    if repository_ctx.attr.target_constraints == None and repository_ctx.attr.exec_constraints == None:
```
Comment (AleksanderGondek) [inline nixpkgs/nixpkgs.bzl:895]: You can use `if not variable and not other_variable` to cover uscases of empty strings and null-like values.

Comment (tek) [reply nixpkgs/nixpkgs.bzl:895]: empty list is not valid for specification by the consumer?

Comment (AleksanderGondek) [reply nixpkgs/nixpkgs.bzl]: This is just a soft-ball comment, Starlark is Pythonesque and  I think it is generally a bad smell to use `== None` in Python .

**Pull request merged**